nonos/main.py

```diff
@@ -680,503 +703,201 @@ def plot(
-        format as bidirectional=False.
-        The format in this case is:
 
-                                (np.array([[x],[y]]),np.array([[x],[y]]))
+            extent = range_converter(
+                extent,
+                abscissa=xi,
+                ordinate=yi,
+            )
 
-        This format is a little bit more complicated, and the best way to manipulate it is with iterators.
-        For example, if you want to plot the streams computed with bidirectional=True, you can do:
+        else:
+            # convert 1D coordinates arrays (self.coord)
+            # into 2D coordinates arrays (coordgrid) via meshgrid,
+            # using (plane[0],plane[1]) for the projection plane
+            # and self.DEFAULT_POINT for the 3d (plane[2]) dimension
+            coordgrid = meshgridFromPlane(
+                self.coord, plane[0], plane[1], self.DEFAULT_POINT
+            )
+            coordgrid = np.array(coordgrid, dtype=object)[np.array(plane) - 1]
+
+            # (coord[0]=R,coord[1]=phi by default)
+            # then, depending on the geometry,
+            # we transform these 2D coordinates arrays
+            transform = func_proj(
+                coordgrid[0],
+                coordgrid[1],
+                coordgrid[2],
+            )
 
-        stream = get_stream(x0,y0)
-        ax.plot(stream[0][0],stream[0][1]) #Forward
-        ax.plot(stream[1][0],stream[1][1]) #Backward
+            im = ax.pcolormesh(
+                transform[plane[0] - 1],
+                transform[plane[1] - 1],
+                data,
+                cmap=cmap,
+                vmin=vmin,
+                vmax=vmax,
+                **karg,
+            )
 
-        """
+            extent = range_converter(
+                extent,
+                abscissa=transform[plane[0] - 1],
+                ordinate=transform[plane[1] - 1],
+            )
 
-        if bidirectional:
-            s0 = self.get_stream(
-                vx,
-                vy,
-                x0,
-                y0,
-                reverse=False,
-                bidirectional=False,
-                nmax=nmax,
-                maxlength=maxlength,
+        if self.init.config["grid"]:
+            ax.plot(
+                transform[plane[0] - 1], transform[plane[1] - 1], c="k", linewidth=0.07
             )
-            s1 = self.get_stream(
-                vx,
-                vy,
-                x0,
-                y0,
-                reverse=True,
-                bidirectional=False,
-                nmax=nmax,
-                maxlength=maxlength,
+            ax.plot(
+                transform[plane[0] - 1].transpose(),
+                transform[plane[1] - 1].transpose(),
+                c="k",
+                linewidth=0.07,
             )
-            return (s0, s1)
 
-        l = 0
-        x = [x0]
-        y = [y0]
+        logging.debug(f"xmin: {extent[0]}")
```

Comment:
fstrings are bad practice in logging entries. This is how you format strings in logging:
```suggestion
        logging.debug("xmin: %d", extent[0])
```
(I can explain in more depth if you wish)